docs/introduction/architecture.md

```diff
@@ -0,0 +1,55 @@
+# Tendermint Architectural Overview
+_November 2019_
+
+Over the next few weeks, @brapse, @marbar3778 and I (@tessr) are having a series of meetings to go over the architecture of Tendermint Core. These are my notes from these meetings, which will either serve as an artifact for onboarding future engineers; or will provide the basis for such a document.
+
+## Communication
+
+There are three forms of communication (e.g., requests, responses, connections) that can happen in Tendermint Core: *internode communication*, *intranode communication*, and *client communication*. 
+- Internode communication: Happens between a node and other peers. This kind of communication happens over TCP or HTTP. More on this below. 
+- Intranode communication: Happens within the node itself (i.e., between reactors or other components). These are typically function or method calls, or occasionally happen through an event bus. 
+- Client communiation: Happens between a client (like a wallet or a browser) and a node on the network.
+
+### Internode Communication
+
+Internode communication can happen in two ways:
+1. TCP connections through the p2p package
+    - Most common form of internode communication
+    - Connections between nodes are persisted and shared across reactors, facilitated by the switch. (More on the switch below.)
+2. RPC over HTTP 
```

Comment:
Why isn't this considered Client Communication?

Reply:
Oh! I guess I assumed that client communication _also_ happens this way, but my understanding is that nodes can also talk to one another over HTTP. Is that true?